vm/src/obj/objstr.rs

```diff
@@ -429,6 +430,29 @@ impl PyString {
         }
     }
 
+    #[pymethod(name = "__mod__")]
+    fn modulo(&self, values: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        let format_string_text = &self.value;
+
+        match CFormatString::from_str(format_string_text.as_str()) {
+            Ok(format_string) => perform_clike_format(vm, format_string, values.clone()),
+            Err(err) => match err.typ {
+                CFormatErrorType::UnsupportedFormatChar(c) => Err(vm.new_value_error(format!(
+                    "unsupported format character '{}' ({:#x}) at index {}",
+                    c, c as u8, err.index
```

Comment:
In the `c as u8` we assume ascii, which might not be the case. Consider `c as u32` at least. Is there a way to get the unicode code point from a char as `usize`? This would be nice!